[READY][:(] code improvements and fixes to fugitive events

tralezab opened the pull request:
long time coming really, can't add anything new without fixing these problems

Fixes #43945

tralezab committed:
testing commit, code improvements

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into feels-like-i-didn't-even-try

tralezab committed:
a lot of the fixes minus ship spawning

fix names

tralezab commented on code/datums/shuttles.dm:
```diff
@@ -554,6 +558,14 @@
+	suffix = "space_cop"
+	name = "Police Spacevan"
+
+/datum/map_template/shuttle/hunter/russian
```
there's still something wrong with these that makes the map template not load. dnm until i work that out

AnturK replied:
So is this in working state now or still needs something ?

tralezab replied:
got it, finally...

Fikou commented:
every single shuttle in the game has a docking port, those fugitive ones don't, put docking ports in

tralezab committed:
Merge branch 'master' into feels-like-i-didn't-even-try

tralezab committed:
confusion will be my epitaph

AnturK commented on code/modules/events/fugitive_spawning.dm:
```diff
@@ -12,27 +12,29 @@
+	var/list/possible_spawns = list()//Some xeno spawns are in some spots that will instantly kill the refugees, like atmos
+	for(var/turf/X in GLOB.xeno_spawn)
+		if(istype(get_area(X), /area/maintenance))
+			possible_spawns += X
```
Is there always one, should probably fail if there isn't. And since you know these are turfs you don't need to use get_area.

AnturK replied:
I think you meant X.loc in both lines here

tralezab replied:
if X.loc would be the turf's area than the above check needs to be that, but this line at least should stay because it's adding a turf to the possible spawns

ExcessiveUseOfCobblestone replied:
turf's loc is area

AnturK commented on code/modules/antagonists/fugitive/old_god.dm:
```diff
@@ -2,24 +2,31 @@
 	return ..()
 
+/mob/camera/yalp_elor/Destroy()
+	. = ..()
```
move this to the end of the proc

AnturK commented on code/modules/antagonists/fugitive/old_god.dm:
```diff
@@ -28,26 +35,26 @@
+			to_chat(src, "<span class='warning'>The Chapel is hallowed ground under a much, MUCH stronger deity, and can't be accessed!</span>")
+		return
+	forceMove(T)
+	Moved(loc, direct)
```
Makes no sense, you just moved so loc is not where you were.

tralezab replied:
i was wondering why that var was nessessary....

tralezab committed:
invisible icon b gone

tralezab committed:
review

tralezab committed:
i want to get off mr bones wild ride, desc

todo: more map instance work, make it actually move, do all this for the russian ship. then and only then it'll be fine

tralezab committed:
removes testing lines

tralezab committed:
test

tralezab committed:
Revert "test"

This reverts commit 786f242ef2982cd659836d755e1240e4541e97a9.

ATH1909 commented:
@tralezab was I supposed to spawn with no equipment as a fugitive (in a group of three)? The original PR says that us prisoners were supposed to get shivs.

Also, we had no objectives. Was that intended as well?

tralezab commented:
The objective is surviving to the end of the round with the state of the team's freedom being how much you win. I should specify this in an individual objective as well.

tralezab committed:
finish line

tralezab commented:
Actually let me very quickly handle the review. I want to be positive this wont break

tralezab committed:
begone, badcode!

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into feels-like-i-didn't-even-try

tralezab committed:
there is no finish line. there is no escape. more fixes.

tralezab committed:
okay, for real, really done

tralezab commented:
okay really for real done wow

tralezab committed:
removes delete_on_mind_deletion

ShizCalev commented on _maps/shuttles/hunter_russian.dmm:
```diff
@@ -95,410 +70,431 @@
-/obj/machinery/door/airlock/security/glass,
+"r" = (
+/obj/effect/turf_decal/arrows{
+	icon_state = "arrows";
```
DV

ShizCalev commented on _maps/shuttles/hunter_space_cop.dmm:
```diff
@@ -0,0 +1,263 @@
+	pixel_y = 26
+	},
+/obj/effect/turf_decal/stripes{
+	icon_state = "warningline";
```
DV

ShizCalev commented on _maps/shuttles/hunter_space_cop.dmm:
```diff
@@ -0,0 +1,263 @@
+/area/shuttle/hunter)
+"af" = (
+/obj/effect/turf_decal/stripes{
+	icon_state = "warningline";
```
DV

ShizCalev commented on _maps/shuttles/hunter_space_cop.dmm:
```diff
@@ -0,0 +1,263 @@
+/area/shuttle/hunter)
+"ah" = (
+/obj/effect/turf_decal/loading_area{
+	icon_state = "loadingarea";
```
DV
